pkg/keycloak/jsonInjector.go

```diff
@@ -0,0 +1,227 @@
+
+	// Path in the template is .parameters
+	err := tpl.ArrayAppend(ssoPlugins.Data(), "parameters")
+
+	if err != nil {
+		return nil, err
+	}
+
+	return tpl, nil
+}
+
+// Injects a volume of type emptyDir into the RHSSO pod that is used to store the
+// installed plugins
+func (j *JsonInjectorImpl) InjectVolume(tpl *gabs.Container) (*gabs.Container, error) {
+	pluginsVolume := gabs.New()
+	pluginsVolume.Set(j.PluginsVolumeName, "name")
+	pluginsVolume.Set(gabs.New().Data(), "emptyDir")
+
+	// Path in the template is .objects[4].spec.template.spec.volumes
+	deploymentConfig, err := j.LookupDeploymentConfig(tpl)
+	if err != nil {
+		return nil, err
+	}
+
+	err = deploymentConfig.S("spec").S("template").S("spec").ArrayAppend(pluginsVolume.Data(), "volumes")
+	if err != nil {
+		return nil, err
+	}
+
+	return tpl, nil
+}
+
+// Injects a volume mount into the RHSSO container that points to the installed
+// plugins. It has to be mounted at the path where keycloak loads plugins at startup
+func (j *JsonInjectorImpl) InjectVolumeMount(tpl *gabs.Container) (*gabs.Container, error) {
+	volumeMount := gabs.New()
+	volumeMount.Set(j.PluginVolumeMount, "mountPath")
+	volumeMount.Set(j.PluginsVolumeName, "name")
+	volumeMount.Set(false, "readonly")
+
+	// Path in the template is .objects[4].spec.template.spec.containers[0].volumeMounts
+	deploymentConfig, err := j.LookupDeploymentConfig(tpl)
+	if err != nil {
+		return nil, err
+	}
+
+	container, err := j.LookupContainer(deploymentConfig.S("spec").S("template").S("spec"))
+	if err != nil {
+		return nil, err
+	}
+
+	container.ArrayAppend(volumeMount.Data(), "volumeMounts")
+	if err != nil {
+		return nil, err
+	}
+
+	return tpl, nil
+}
+
+// Injects the init container into the RHSSO pod that will copy all plugin binaries from
+// the init container into the mounted volume
+func (j *JsonInjectorImpl) InjectInitContainer(tpl *gabs.Container) (*gabs.Container, error) {
+	// Init container base object
+	initContainer := gabs.New()
+
+	// Init container env section
+	initContainerEnv := gabs.New()
+	initContainerEnv.Set(j.PluginsEnvVarName, "name")
+	initContainerEnv.Set(fmt.Sprintf("${%s}", j.PluginsEnvVarName), "value")
+
+	// Init container volume mounts
+	initContainerMount := gabs.New()
+	initContainerMount.Set(j.InitContainerMount, "mountPath")
+	initContainerMount.Set(j.PluginsVolumeName, "name")
+	initContainerMount.Set(false, "readonly")
+
+	initContainer.ArrayAppend(initContainerEnv.Data(), "env")
+	initContainer.ArrayAppend(initContainerMount.Data(), "volumeMounts")
+
+	// Init container name and image
+	initContainer.Set(j.InitContainerImage, "image")
+	initContainer.Set(j.InitContainerName, "name")
+
+	// Path in the template is .objects[4].spec.template.spec.initContainers
+	deploymentConfig, err := j.LookupDeploymentConfig(tpl)
+	if err != nil {
+		return nil, err
+	}
+
+	err = deploymentConfig.S("spec").S("template").S("spec").ArrayAppend(initContainer.Data(), "initContainers")
```

Comment:
Do you think a check to see if the init container is already in the template is worth while, or not necessary?